teps/0130-pipeline-level-build-cache.md

```diff
@@ -0,0 +1,557 @@
+(There are a few problems with it: the same deployment and service will be used for each instance of the PipelineRun, and multiple instances can interfere with each other;
+the deployment name and its volume mounts are hard-coded into the Task spec but the yaml is passed in via a workspace.)
+
+A better approach might be to use a Task similar to the Catalog Task
+[kubectl-deploy-pod](https://github.com/tektoncd/catalog/blob/5f6fc936854fa05ac710be12ecb8b43c8f37f742/task/kubectl-deploy-pod/0.1/kubectl-deploy-pod.yaml),
+and modify it to create a service and a deployment (or use it twice, once for the service and once for the deployment), or to write custom code and build it into
+a Task used specifically for creating a Docker daemon.
+
+### Performance
+
+<!--
+(optional)
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+- What impact does this change have on the start-up time and execution time
+of TaskRuns and PipelineRuns?
+- What impact does it have on the resource footprint of Tekton controllers
+as well as TaskRuns and PipelineRuns?
+-->
+
+### Security
+
+- We will need to make sure other PipelineRuns (or anything else on the cluster) is not able to use the service
+created for the Pipeline-level sidecar
+- TODO: Investigate what implications, if any, this proposal has for provenance
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Tekton ecosystem. Consider including folks that also work outside the WGs
+or subproject.
+- How will security be reviewed and by whom?
+- How will UX be reviewed and by whom?
+-->
+
+### Drawbacks
+
+- If docker TLS certs are used, they must now be bound in a persistent volume claim to share
+between multiple Tasks, rather than an emptyDir volume to share between multiple steps.
+This is a pretty heavyweight option for a relatively small amount of data.
+However, this is more an artifact of the limited options Pipelines currently has for sharing
+data between Tasks, which should be treated as a separate problem.
+
+- Sidecar configurations will likely be copy-pasted between Pipelines.
+
+## Alternatives
+
+### Improve our documentation
+
+Instead of building new features, we could improve our documentation to help users better understand how to build a docker daemon
+into their PipelineRun. See [user experience](#user-experience) for an example of what this would look like.
+
+### Catalog Tasks for docker daemon
+
+We could add Tasks to the verified catalog (or to a community catalog) for spinning up and tearing down a docker daemon (or more generically, a deployment with a service).
+This would prevent users from having to write their own daemon creation/teardown tasks, but they would still have to provide their kube credentials
+and use a service account with rolebindings that allow it to create services and deployments.
```

Comment:
Yes and yes. For me those are the valid, most important and "today doable" action to take. From these, their usage and feedback from it, we may want to enhance `v1` or define `v2` better.